src/App.js

```diff
@@ -1,12 +1,53 @@
 import React, { Component } from 'react';
+
+import HeroData from "./utils/heroes.json";
+
 import HeroCard from './components/HeroCard';
 import './App.scss';
 
+
+const arrayToMap = (array) => {
+  if (!Array.isArray(array)) {
+      return {};
+  }
+
+  return array.reduce((aggr, obj) => {
+      return {
+          ...aggr,
+          [obj.id]: obj
+      };
+  }, {});
+};
+
+//Render our hero cards for the app to use. 
+const heroRender = (heroList) => {
+
+  return (
+    Object.keys(heroList).map((key) => <HeroCard key={heroList[key].id} heroData={heroList} hero={heroList[key]} counters={heroList[key].counters}/>)
+  )
+}
+
+
 class App extends Component {
+  state = {
+    "hero": null,
+  }
+
+  componentWillMount() {
+    console.log(arrayToMap(HeroData))
+    this.setState({
+        "hero": arrayToMap(HeroData),
+    })
+
+    
+}
+
   render() {
+    
+    let { hero } = this.state;
     return (
       <div>
-        <HeroCard />
+    {!hero ? <p>Loading</p> : heroRender(this.state.hero)}
```

Comment:
It's best to avoid putting logic in your JSX. This is a bit more verbose, but consider the readability:

```javascript
render() {
    const { hero } = this.state;
    const placeholder = <p>Loading</p>;
    const content = (hero) ? heroRender(hero) : placeholder;
    return (
        <div>{content}</div>
    );
}
```

Reply:
I like this. 

I was also thinking of doing  different returns? 

Thinking of 

`
render() {
const { hero } = this.state;
    const placeholder = <p>Loading</p>;
if(hero) {
    return ( heroRender(hero))
}`

    return placeholder;
}

What do you think?

Reply:
I've written functions like that in the past, so your thinking isn't wrong. It's a habit I broke, though, because over time code will get added between the two return statements, and eventually someone will go to make a change and not see the return statement above it, and not realize their change isn't going to happen to data that's already been returned.

Or they'll change what gets returned at the bottom and not realize they need to change it anywhere else.

Reply:
Oh! And I should add there's an exception to this generalization...

In the event that a function should suffer a fatal condition, it's acceptable to return early in the function. In an ideal world, you'd throw an Exception, but in practice it's more common to do things like return false, <1, [] or {}, unless you're working for a shop that's diligent about its try{}catch(){} blocks.

So if you're expecting a parameter to be a string, and you get an integer, you might return immediately, since you know whatever you were expected to do isn't going to work. But that doesn't violate what I said above, because you'll always want to return that error condition, even if someone adds mutations to the code, below it.